business/checkers/destinationrules/traffic_policy_checker.go

```diff
@@ -0,0 +1,102 @@
+package destinationrules
+
+import (
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+)
+
+type TrafficPolicyChecker struct {
+	DestinationRules []kubernetes.IstioObject
+	MTLSDetails      kubernetes.MTLSDetails
+}
+
+func (t TrafficPolicyChecker) Check() models.IstioValidations {
+	validations := models.IstioValidations{}
+
+	// When mTLS is not enabled, there is no validation to be added.
+	if !t.hasEnablingmTLSDR() {
+		return validations
+	}
+
+	// Check whether DRs override mTLS.
+	for _, dr := range t.DestinationRules {
+		if !hasTrafficPolicy(dr) || !hasmTLSSettings(dr) {
+			check := models.Build("destinationrules.trafficpolicy.notlssettings", "spec/trafficPolicy")
+			key := models.BuildKey(DestinationRulesCheckerType, dr.GetObjectMeta().Name)
+			validation := buildDestinationRuleValidation(dr, check, true)
+
+			if _, exists := validations[key]; !exists {
+				validations.MergeValidations(models.IstioValidations{key: validation})
+			}
+		}
+	}
+
+	return validations
+}
+
+func (t TrafficPolicyChecker) hasEnablingmTLSDR() bool {
+	enablesTLS := false
+
+	for _, dr := range t.MTLSDetails.DestinationRules {
+		enablesTLS = enablesmTLS(dr)
+		if enablesTLS {
+			break
+		}
+	}
+
+	return enablesTLS
+}
+
+func hasTrafficPolicy(dr kubernetes.IstioObject) bool {
+	_, trafficPresent := dr.GetSpec()["trafficPolicy"]
+	return trafficPresent
+}
+
+func hasmTLSSettings(dr kubernetes.IstioObject) bool {
+	return enablesAnyTLS(dr) || enablesPortTLS(dr)
+}
+
+// enablesPortTLS returns true when there is one port that specifies any TLS settings
+func enablesPortTLS(dr kubernetes.IstioObject) bool {
```

Comment:
I'm not checking other code yet.
as we have "hasTrafficPolicy" kinda syntax it sounds to me interesting to rename this to "isPortTLSEnabled".
When I read enablesPortTLS I was expecting some side effect on the dr.